File: cpp-ami/Client-end.cpp
#include <Ice/Ice.h>
#include "factorial.h"

using namespace std;
using namespace Ice;

class Client: public Ice::Application {
  int run(int argc, char* argv[]) {
    ObjectPrx proxy = communicator()->stringToProxy(argv[1]);
    Example::MathPrx math = Example::MathPrx::checkedCast(proxy);

    Ice::AsyncResultPtr async_result = math->begin_factorial(atoi(argv[2]));
    cout << "that was an async call" << endl;

    cout << math->end_factorial(async_result) << endl;
    return 0;
  }
};


int main(int argc, char* argv[]) {
  if (argc != 3) {
    cerr << "usage: " << argv[0] << "<server> <value>" << endl;
    return 1;
  }

  Client app;
  return app.main(argc, argv);
}

File: java-ami/Client_end.java
public class Client_end extends Ice.Application {
  public int run(String[] args) {
    Ice.ObjectPrx proxy = communicator().stringToProxy(args[0]);
    Example.MathPrx math = Example.MathPrxHelper.checkedCast(proxy);

    int value = Integer.parseInt(args[1]);
	 Ice.AsyncResult async_result = math.begin_factorial(value);
    System.out.println("that was an async call");

    System.out.println(math.end_factorial(async_result));
    return 0;
  }

  static public void main(String[] args) {
    if (args.length != 2) {
      System.err.println(appName() + ": usage: <server> <value>");
      return;
    }

    Client_end app = new Client_end();
    app.main("Client", args);
  }
}
